### PR TITLE
drivers: sensors: ens210: fix fractional part calculation

### PR DESCRIPTION
**Pull request by alexanderwachter**
This commit fixes the calculation of the fractional part
for both values.

Signed-off-by: Alexander Wachter <alexander.wachter@student.tugraz.at>

### CODE REVIEW

File: drivers/sensor/ens210/ens210.c [hunk 1]
```diff
@@ -107,13 +107,13 @@ static int ens210_channel_get(struct device *dev,
 		temp_frac -= 273150000;
 
 		val->val1 = temp_frac / 1000000;
-		val->val2 = temp_frac - val->val1;
+		val->val2 = temp_frac - val->val1 * 1000000;
```
Comment (avisconti) [inline drivers/sensor/ens210/ens210.c]: why not doing:

  val->val2 = temp_frac % 1000000;

This is basically what all the drivers are doing.
Does it make sense to you?

File: drivers/sensor/ens210/ens210.c [hunk 2]
```diff
@@ -107,13 +107,13 @@ static int ens210_channel_get(struct device *dev,
...
 		break;
 	case  SENSOR_CHAN_HUMIDITY:
 		humidity_frac = sys_le16_to_cpu(drv_data->humidity.val) *
 				(1000000 / 512);
 		val->val1 = humidity_frac / 1000000;
-		val->val2 = humidity_frac - val->val1;
+		val->val2 = humidity_frac - val->val1 * 1000000;
```
Comment (avisconti) [inline drivers/sensor/ens210/ens210.c]: same here

**Review by avisconti — CHANGES_REQUESTED** [review]

File: drivers/sensor/ens210/ens210.c (hunk 1 above)
Comment (alexanderwachter) [reply drivers/sensor/ens210/ens210.c]: Sure

**Review by avisconti — APPROVED** [review]
LGTM!
Anyway, have you re-tested it?

**Comment by alexanderwachter** [comment]
> Anyway, have you re-tested it?

Yes. Works fine!

**Comment by alexanderwachter** [comment]
Rebased.

**Review by MaureenHelm — APPROVED** [review]

**Pull request merged**